### ISSUE TITLE
Support SwiftPods

### ISSUE THREAD

**Issue by neonichu** [CocoaPods/cocoapods-packager#56]
Eventually, we will want to be able to build dynamic frameworks using the packager. 

The goal should remain that the dependencies are vendored with mangled symbols by default, so that the generated framework is suitable for easy distribution. As dynamic frameworks are more reasonable in terms of users providing transitive dependencies themselves, there could also be an option for producing leaving the dependencies out completely.

In general, dependencies need to be build as static libraries, so that they can be lipo'ed into the dynamic framework. Xcode will take care of public headers and resources for us and Swift Pods gives us a generated module map already.

**Comment by kcoleman731** [comment]
@neonichu found this issue today and would love to know if there has been any progress on dynamic framework packaging?

If not, I'd be happy to start working on a PR to add support for this. I pulled down the project earlier this week to get familiar with the source. Any guidance you can give as to how the dynamic frameworks should to be assembled would be much appreciated.

**Comment by segiddins** [comment]
Iirc there hasn't been any progress on this front yet

**Comment by neonichu** [comment]
Yah, there has been pretty much zero progress, because I didn't need it, yet. 

There will actually be two major parts:
1. support packaging dynamic frameworks
2. support mangling Swift dependencies

which might be better off being split into two different issues. 

@kcoleman731 In any case, a PR would be greatly appreciated, because I probably won't be getting around to this myself too soon.

My approach would be building the Pod being asked to package as a dynamic framework using `use_frameworks!` and building the dependencies as static libraries, just as before.

**Comment by kcoleman731** [comment]
@neonichu Sounds great, already working on the pr. 

I already have the separate pieces building properly. I'm a bit stumped on how you combine the dynamic framework + the static dependencies into one build artifact however. In your first comment above, you recommended using `lipo`. If I do this, I get the following error msg. 
```
[<dynamic_framework_name> and <staticlibs.a>] have the same architectures (armv7) and can't be in the same fat output file
```

Any suggestions?

**Comment by neonichu** [comment]
@kcoleman731 ah, I thought `lipo` would work — alternatively, you could simply try linking the dynamic framework against the static libs.

**Comment by kcoleman731** [comment]
@neonichu got it. Moving on to that approach.....I was just looking through the `Xcodeproj` gem. I figured there might be some API to specify linked libraries....not finding any however. Am I missing something obvious?

**Comment by neonichu** [comment]
Implemented by #99

**Comment by Kjens93** [comment]
@neonichu's [comment](#issuecomment-145156279) appears to be incorrect. It would appear that this was NOT implemented by #99. See bug #115.
